src/test/kotlin/oop/Memento/MementoTest.kt

```diff
@@ -0,0 +1,43 @@
+package oop.Memento
+
+import com.natpryce.hamkrest.Matcher
+import com.natpryce.hamkrest.assertion.assert
+import org.junit.Test
+
+class MementoTest {
+
+  fun <T> `is`(list: List<T>): Matcher<List<T>> = Matcher(List<T>::equals, list)
+  fun <T> `is`(list: Memento<T>): Matcher<Memento<T>> = Matcher(Memento<T>::equals, list)
+
+  @Test
+  fun `memento state should be equal to last originator state`() {
+    val originator = Originator(emptyList<Int>())
+    originator.state = listOf(1,2,3)
+
+    val memento = originator.saveToMemento()
+
+    assert.that(memento.state, `is`(memento.state))
```

Comment:
In this case this method will always be true, isn't it. Because here you are comparing memento state with the same memento state, not with the originator. 🤔